lib/datastruct/patricia.c

```diff
@@ -144,7 +144,7 @@ compare(const struct pnode * n, const uint8_t * key, size_t keylen,
 	for (i = 0; i < n->slen; i++) {
 		/* Is the key a prefix of the node? */
 		if (keylen == i) {
-			*mlen = i;
+			*mlen = i & 0xff;
```

Comment:
Hey, I never claimed that I was consistent!  In this case we're iterating between 0 and a `uint8_t` value though, which makes it more natural to use a `uint8_t` value... even though we're eventually using it as an array index (which would normally suggest making it a `size_t`).

*shrug*